timeflux_dsp/nodes/spectral.py

```diff
@@ -63,36 +64,50 @@ def __init__(self, fs=1.0, nfft=None, return_onesided=True):
                                               If `False` return a two-sided spectrum.
                                               (Note that for complex data, a two-sided spectrum is always returned.)
                                               Default: `True`.
+                interpolation (bool|dict): How to interpolate NaN values in the input using Pandas interpolate.
         """
 
-        self._fs = fs
-        self._nfft = nfft
+        self._fixed_rate = rate
+        self._rate = rate
+        self._nfft = None if nfft is None else int(nfft)
+        self._fixed_nfft = self._nfft
```

Comment:
See my main comment. I'm not sure this is relevant to allow for a changing nfft.

Reply:
Agreed, removed.